codegen/src/TypeMapper.scala

```diff
@@ -1,35 +1,56 @@
 package besom.codegen
 
+import besom.codegen.Config.ProviderConfig
+
 import scala.util.matching.Regex
 import scala.meta._
 import scala.meta.dialects.Scala33
 import besom.codegen.metaschema._
 
+case class TypeToken private (provider: String, module: String, `type`: String, raw: String) {
+  def asString: String          = s"${provider}:${module}:${`type`}"
+  override def toString: String = asString
+}
+
+object TypeToken {
+  private val typeTokenFmt: String    = "(.*):(.*)?:(.*)" // provider:module:type
+  private val typeTokenPattern: Regex = ("^" + typeTokenFmt + "$").r
+
+  def apply(typeToken: String)(implicit config: ProviderConfig): TypeToken = typeToken match {
+    case typeTokenPattern(p, m, t) => {
+      val module =  m match {
+        case "" if config.ensureIndexModule => "index"
```

Comment:
Shouldn't it be `enforceIndexModule`? `ensure` would rather suggest crashing if the requirement is not fulfilled. Or maybe `inferIndexModule` would be even better?

Reply:
ensure - make certain that (something) will occur or be the case

there is no inference, just making sure the index is present

Reply:
I'll make a change to `enforceIndexModule` to avoid iny confusion, thank you